src/meta/processors/jobMan/AdminJobProcessor.cpp

```diff
@@ -0,0 +1,115 @@
+/* Copyright (c) 2019 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include <string>
+#include <vector>
+#include "common/base/StatusOr.h"
+#include "meta/processors/jobMan/AdminJobProcessor.h"
+#include "meta/processors/jobMan/JobManager.h"
+#include "meta/processors/jobMan/JobDescription.h"
+
+namespace nebula {
+namespace meta {
+
+void AdminJobProcessor::process(const cpp2::AdminJobReq& req) {
+    Status status;
+    std::vector<std::string> result;
+    std::stringstream oss;
+    for (auto& p : req.get_paras()) { oss << p << " "; }
+    LOG(INFO) << __PRETTY_FUNCTION__ << " paras=" << oss.str();
+
+    JobManager* jobMgr = JobManager::getInstance();
+    switch (req.get_op()) {
+        case nebula::meta::cpp2::AdminJobOp::ADD:
+        {
+            auto jobDesc = jobMgr->buildJobDescription(req.get_paras());
+            if (jobDesc.ok()) {
+                result.emplace_back(std::to_string(jobMgr->addJob(jobDesc.value())));
+            } else {
+                status = jobDesc.status();
+            }
+            break;
+        }
+        case nebula::meta::cpp2::AdminJobOp::SHOW_All:
+        {
+            auto r = jobMgr->showJobs();
+            if (r.ok()) {
+                result.swap(r.value());
+            } else {
+                status = r.status();
+            }
+            break;
+        }
+        case nebula::meta::cpp2::AdminJobOp::SHOW:
+        {
+            if (req.get_paras().empty()) {
+                status = Status::SyntaxError("show job needs para");
+            }
+            int iJob = atoi(req.get_paras()[0].c_str());
+            if (iJob == 0) {
+                break;
+            }
+            auto r = jobMgr->showJob(iJob);
+            if (r.ok()) {
+                result.swap(r.value());
+            } else {
+                status = r.status();
+            }
+            break;
+        }
+        case nebula::meta::cpp2::AdminJobOp::STOP:
+        {
+            if (req.get_paras().empty()) {
+                status = Status::SyntaxError("stop job needs para");
+            }
+            int iJob = atoi(req.get_paras()[0].c_str());
+            if (iJob == 0) {
+                break;
+            }
+            status = jobMgr->stopJob(iJob);
+            if (status.ok()) {
+                result.emplace_back("job " + std::to_string(iJob) + " stopped");
+            }
+            break;
+        }
+        case nebula::meta::cpp2::AdminJobOp::BACKUP:
```

Comment:
What's the meaning for "Backup"?  

It is called by users directly?

Reply:
> What's the meaning for "Backup"?

 In case of the result of "show jobs" grow up too much to read.  
"backup" jobs will not show in "show jobs"  again.

> It is called by users directly?

yes... Do you mean need some authentication thing here ? I got the point this will user B "backup" user A's job.

Reply:
I mean do we really need this feature. For expired jobs,  we could compact them asynchronously

Reply:
sometimes I want to "backup" those trash jobs immediately?

Reply:
backup for what?

Reply:
If there isn't any user called interface here. 
let's say there are 30, or more trash jobs, and I set the background backup job period for 3days. 
each time I type "show jobs", I got that trash jobs result. 
the only way I can get rid of it is to wait 3 days.

Reply:
When "show jobs",  we should filter the expired jobs